### ISSUE TITLE
Flask-User YANKED pypi

### ISSUE THREAD

**Issue by maucm92** [lingthio/Flask-User#341]
Hello,
First of all, congratulations for the excellent work on this project.
We recently detected that the project on [PyPi](https://pypi.org/) is listed as "Yanked".
I have been reported that doing pip install Flask-User does not work due to the state of the project and the version number must be added to be able to install. 
On the other hand, a warning message is observed during the installation and a reason why the project is marked as Yanked is not indicated.

![yanked](https://user-images.githubusercontent.com/39063659/154959061-2eeb3bbf-69a4-481f-bf7c-f88408f6bfad.png)

My questions are:

1. What can we do to avoid the mentioned problems and continue using the module normally?
2. Since when is the project marked as Yanked?
3. Is this due to a bug in the module?
4. Or is it because the project is no longer in active development and will be removed or deprecated?
5. Should we look for an alternative project?

Thank you very much

Greetings,
Mauro

**Comment by charlbury** [comment]
Same questions from me. As additional information, if I do a 

```pip install flask-user```

I get the following error:

```
ERROR: Could not find a version that satisfies the requirement flask-user (from versions: 0.3, 0.3.1, 0.3.2, 0.3.3, 0.3.4, 0.3.5, 0.3.6, 0.3.7, 0.3.8, 0.4.0, 0.4.1, 0.4.2, 0.4.3, 0.4.4, 0.4.5, 0.4.6, 0.4.7, 0.4.8, 0.4.9, 0.5.0, 0.5.1, 0.5.2, 0.5.3, 0.5.4, 0.5.5, 0.6, 0.6.1, 0.6.2, 0.6.3, 0.6.4, 0.6.5, 0.6.6, 0.6.7, 0.6.8, 0.6.9, 0.6.10, 0.6.12, 0.6.13, 0.6.14, 0.6.15, 0.6.16, 0.6.17, 0.6.19, 0.6.20, 0.6.21, 1.0.1.1, 1.0.1.2, 1.0.1.3, 1.0.1.4, 1.0.1.5, 1.0.2.0, 1.0.2.1, 1.0.2.2)
ERROR: No matching distribution found for flask-user
```

**Comment by jgerry2002** [comment]
Yes I've seen this same issue. Not sure what happened to this version.

**Comment by peterstavrou** [comment]
If anyone knows the answer to this, I would also love to know.
Perhaps its unsafe to use this package for now?

**Comment by amanve** [comment]
[It's not updated for recent ](https://github.com/lingthio/Flask-User/issues/180)

**Comment by jgerry2002** [comment]
I'm guessing the maintainer has lost interest in this project. There has been no activity since 2020 in his github account. If he is out there hopefully he will respond soon.

**Comment by amanve** [comment]
You can fix this manually -
1. Clone the repo - https://github.com/lingthio/Flask-User.git
2. upgrade your "pip" - py -m pip install --upgrade pip
3. upgrade "build" - py -m pip install --upgrade build
4. browse to the folder location of the repo cloned
5. then build the package - py -m build
6. then go to 'dist' folder created there would be a *filename*.whl file
7. run pip install *filename*.whl
8. at last *install pip install wtforms[email]* for email validators.
 Alternatively install from here -
[Flask_User-1.0.2.3-py2.py3-none-any.zip](https://github.com/lingthio/Flask-User/files/8266816/Flask_User-1.0.2.3-py2.py3-none-any.zip)

**Comment by dave-c-vt** [comment]
> I'm guessing the maintainer has lost interest in this project. There has been no activity since 2020 in his github account. If he is out there hopefully he will respond soon.

oof. hopefully guy is ok, otherwise, we should probably fork and re-pypi this project under a different moniker?

**Comment by jgerry2002** [comment]
> > I'm guessing the maintainer has lost interest in this project. There has been no activity since 2020 in his github account. If he is out there hopefully he will respond soon.
> 
> oof. hopefully guy is ok, otherwise, we should probably fork and re-pypi this project under a different moniker?

It may be a good idea to consider since we've seen no activity if it makes sense to keep this project going.

**Comment by peterstavrou** [comment]
Can always use [flask-security-too](https://github.com/Flask-Middleware/flask-security/) as an alternative.

**Comment by yejuns** [comment]
I also had the same problem, finally after updating the pip version by the command 'python -m pip install --upgrade pip', the installation was successful

**Comment by johan-sorman** [comment]
Can't install it either.

**Comment by GitauHarrison** [comment]
> Same questions from me. As additional information, if I do a
> 
> `pip install flask-user`
> 
> I get the following error:
> 
> ```
> ERROR: Could not find a version that satisfies the requirement flask-user (from versions: 0.3, 0.3.1, 0.3.2, 0.3.3, 0.3.4, 0.3.5, 0.3.6, 0.3.7, 0.3.8, 0.4.0, 0.4.1, 0.4.2, 0.4.3, 0.4.4, 0.4.5, 0.4.6, 0.4.7, 0.4.8, 0.4.9, 0.5.0, 0.5.1, 0.5.2, 0.5.3, 0.5.4, 0.5.5, 0.6, 0.6.1, 0.6.2, 0.6.3, 0.6.4, 0.6.5, 0.6.6, 0.6.7, 0.6.8, 0.6.9, 0.6.10, 0.6.12, 0.6.13, 0.6.14, 0.6.15, 0.6.16, 0.6.17, 0.6.19, 0.6.20, 0.6.21, 1.0.1.1, 1.0.1.2, 1.0.1.3, 1.0.1.4, 1.0.1.5, 1.0.2.0, 1.0.2.1, 1.0.2.2)
> ERROR: No matching distribution found for flask-user
> ```

Same issue

**Comment by taylenschmaltz** [comment]
well what do we do now

**Comment by johan-sorman** [comment]
> well what do we do now

I migrated to django framework

**Comment by dave-c-vt** [comment]
I was able to switch to using Flask-Login without any trouble at all.

https://github.com/maxcountryman/flask-login

**Comment by colidyre** [comment]
You can also install from git directly, e.g. for a specific version (here latest tag 1.0.2.2):

```bash
pip install -e git+https://github.com/lingthio/Flask-User.git@v1.0.2.2#egg=Flask-User
```

This will be resolved to:
```bash
$ pip freeze | grep Flask-User
-e git+https://github.com/lingthio/Flask-User.git@e3aafd927e1156fb38d9aa729bc729eebe2721d6#egg=Flask_User
```

**Comment by elouahidiahmed** [comment]
Hello, I faced the same issue and I feel unconfortable installing a package that hasn't been updated for more than 2 years. Is there any alternatives to manage users roles suited for production environnement please?

**Comment by Exordio** [comment]
Well I do not know. It feels like the author is dead. in general, this is a very flawed package, even if it allows user authorization, and a small role model. But everything that is inside, you can write 0.3 of the volume of the entire code.

On top of that, this package only has client sessions - this is a complete bullshit in 2023. It's better to use JWT with client fingerprints to revoke tokens.

**Comment by Chaostheorie** [comment]
As note, the author just seems to have moved on from OSS. You can still find them on LinkedIn regularly posting.
